### PR TITLE
fix: add `rust-check` target to the makefile for deps pre-checking

### PR DESCRIPTION
**Pull request by kayx23**
### Issue Description

Following instruction in [Building APISIX from source](https://apisix.apache.org/docs/apisix/next/building-apisix/) to build APISIX and encountered the following issue related to Rust not installed in the environment:

b2749803-6237-478d-8714-1dbb803049ef

Issue was introduced by this PR about LDAP: https://github.com/apache/apisix/pull/9037

### PR Description

Updated Makefile to check if Rust is installed in the environment before installing other deps. 

### Local Tests

image

### Doc Update
[See here](https://github.com/apache/apisix/pull/9444/commits/f82e9c3de351aa713f05dd69ea6b9cc8f73af19a)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by An-DJ** [comment]
LGTM, cc @bzp2010 @monkeyDluffy6017

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Review by kingluo — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -147,10 +147,17 @@ help:
 	fi
 	@echo
 
+### check-rust : check if Rust is installed in the environment
+.PHONY: check-rust
+check-rust:
+	@if ! which rustc >/dev/null; then \
```
Comment (leslie-tsang) [inline Makefile:153]:
```suggestion
	@if [ ! which rustc >/dev/null ]; then \
```

File: Makefile [hunk 2]
```diff
@@ -147,10 +147,17 @@ help:
...
+		echo "ERROR: Rust is not installed. Please install Rust before continuing." >&2; \
+		exit 1; \
+	fi;
```
Comment (leslie-tsang) [inline Makefile:156]:
```suggestion
	fi;

```
Need two blank line between targets. :)

**Review by kingluo — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -147,10 +147,18 @@ help:
 	fi
 	@echo
 
+### check-rust : check if Rust is installed in the environment
+.PHONY: check-rust
+check-rust:
+	@if [ ! which rustc >/dev/null ]; then \
```
Comment (kayx23) [inline Makefile:153]:
```suggestion
	@if ! [ $(shell command -v rustc) ]; then \
```
proposed by @leslie-tsang

**Review by leslie-tsang — APPROVED** [review]

**Review by kingluo — APPROVED** [review]

**Pull request merged**